Dockerfile

```diff
@@ -0,0 +1,55 @@
+# Use Ubuntu 16.04 LTS
+FROM nvidia/cuda:9.1-runtime-ubuntu16.04
+
+# Prepare environment
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+                    curl \
+                    bzip2 \
+                    ca-certificates \
+                    build-essential \
+                    autoconf \
+                    libtool \
+                    pkg-config \
+                    wget \
+                    libboost-all-dev \
+                    unzip \
+                    libgl1-mesa-dev \
+                    libglu1-mesa-dev \
+                    freeglut3-dev \
+                    mesa-utils \
+                    g++ \
+                    jq \
+                    tar \
+                    zip \
+                    gcc \
+                    make \
+                    python \
+                    zlib1g-dev \
+                    imagemagick \
+                    software-properties-common \
+                    git && \
+    curl -sL https://deb.nodesource.com/setup_10.x | bash - && \
+    apt-get install -y --no-install-recommends \
+      nodejs && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+ADD https://cmake.org/files/v3.11/cmake-3.11.4-Linux-x86_64.sh /cmake-3.11.4-Linux-x86_64.sh
+
+ADD . /tmp/ants/source
+RUN mkdir /opt/cmake \
+  && sh /cmake-3.11.4-Linux-x86_64.sh --prefix=/opt/cmake --skip-license \
+  && ln -s /opt/cmake/bin/cmake /usr/local/bin/cmake \
+    && mkdir -p /tmp/ants/build \
+    && cd /tmp/ants/build \
+    && mkdir -p /opt/ants-latest \
+    && git config --global url."https://".insteadOf git:// \
+    && cmake -DBUILD_TESTING=OFF -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX=/opt/ants-latest /tmp/ants/source \
```

Comment:
I've always been a bit confused for shared vs static. The default ants build is static, which means all that's needed is the binaries at the end of everything.

Reply:
Shared libs make for smaller containers. They are less convenient as you have to worry about having the libraries loaded, but I think in a container this can be solved once. That was my take from https://github.com/ANTsX/ANTs/issues/914

Reply:
> This looks cool, but without cleanup we'll have a huge docker file with all the build dependencies in it for no reason. There's some other dockers for ANTs already out there, this guy is doing the right thing with build-then-copy to avoid having a "thick" container:
> https://github.com/kaczmarj/ANTs-builds

I just split up the Dockerfile into two parts and that definitely makes a smaller image. I wasn't aware of that trick.